packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -127,23 +134,87 @@ Future<void> checkGradleDependencies() async {
+  final String newSettingsRelativeFile = fs.path.relative(newSettingsFile.path);
+  final Status status = logger.startProgress('✏️  Creating `$newSettingsRelativeFile`...',
+      timeout: timeoutConfiguration.fastOperation);
+
+  final String flutterRoot = fs.path.absolute(Cache.flutterRoot);
+  final File deprecatedFile = fs.file(fs.path.join(flutterRoot, 'packages','flutter_tools',
+      'gradle', 'deprecated_settings.gradle'));
+  assert(deprecatedFile.existsSync());
+  // Get the `settings.gradle` content variants that should be patched.
+  final List<String> deprecatedFilesContent = deprecatedFile.readAsStringSync().split(';EOF');
+  bool exactMatch = false;
+  for (String deprecatedFileContent in deprecatedFilesContent) {
+    if (currentFileContent.trim() == deprecatedFileContent.trim()) {
+      exactMatch = true;
+      break;
+    }
+  }
+  if (!exactMatch) {
+    status.cancel();
+    printError('*******************************************************************************************');
+    printError('Flutter tried to create the file `$newSettingsRelativeFile`, but failed.');
+    // Print how to manually update the file.
+    printError(fs.file(fs.path.join(flutterRoot, 'packages','flutter_tools',
+        'gradle', 'manual_migration_settings.gradle.md')).readAsStringSync());
+    printError('*******************************************************************************************');
+    throwToolExit('Please create the file and run this command again.');
+  }
+  // Copy the new file.
+  final String settingsAarContent = fs.file(fs.path.join(flutterRoot, 'packages','flutter_tools',
+      'gradle', 'settings_aar.gradle.tmpl')).readAsStringSync();
+  newSettingsFile.writeAsStringSync(settingsAarContent);
+  status.stop();
+  printStatus('✅ `$newSettingsRelativeFile` created successfully.');
+}
+
 // Note: Dependencies are resolved and possibly downloaded as a side-effect
 // of calculating the app properties using Gradle. This may take minutes.
-Future<GradleProject> _readGradleProject() async {
+Future<GradleProject> _readGradleProject({bool isLibrary = false}) async {
   final FlutterProject flutterProject = FlutterProject.current();
   final String gradle = await _ensureGradle(flutterProject);
   updateLocalProperties(project: flutterProject);
+
+  final FlutterManifest manifest = flutterProject.manifest;
+  final Directory hostAppGradleRoot = flutterProject.android.hostAppGradleRoot;
+
+  if (featureFlags.isPluginAsAarEnabled &&
+      !manifest.isPlugin && !manifest.isModule) {
+    createSettingsAarGradle(hostAppGradleRoot);
+  }
+  if (manifest.isPlugin && isLibrary) {
+    return GradleProject(
+      <String>['debug', 'profile', 'release'],
+      <String>[], // Plugins don't have flavors.
+      flutterProject.directory.childDirectory('build').path,
+    );
+  }
   final Status status = logger.startProgress('Resolving dependencies...', timeout: timeoutConfiguration.slowOperation);
   GradleProject project;
   try {
```

Comment:
The logic in this function could use some explanation in comments. For example, it's hard to tell whether gradle failing under this 'try', and falling back on the default is expected/okay, or whether something is probably going to go wrong later on.